### PR TITLE
Add instruction on removes existing metrics-server before addon upgrade

### PR DESCRIPTION
**Pull request by jenting**
fixes #1172804

Signed-off-by: JenTing Hsiao <jenting.hsiao@suse.com>

### CODE REVIEW

**Review by evrardjp — APPROVED** [review]
LGTM

**Comment by jenting** [comment]
> LGTM

Thanks to your kindly review :bow:

File: adoc/release-4.2.1.adoc [hunk 1]
```diff
@@ -14,8 +14,13 @@
 
 === Metrics Server
 
-The Metrics Server now supports monitoring of *CPU* and *memory* of a pod or node.  You can use commands `kubectl top nodes` get status for node and use `kubectl top pods` to get status for pod.
-This information can also be used and show in be Prometheus or Grafana.
+{productname} now supports the metrics server as an addon to monitors the *CPU* and *memory* of a pod or node.  You can use commands `kubectl top nodes` get status for node and use `kubectl top pods` to get status for pod. Also, with metrics-server default built-in by {productname}, the user could set up a horizontal pod autoscaler (HPA) to auto-scales the number of pods or set up a vertical pod autoscaler (VPA) to auto allocates more or less CPUs and memory resource.
+
+{productname} did not detect existed metrics-server installed by the administrator, so please remove the installed metrics-server manually before upgrade addons.
```
Comment (r0ckarong) [inline adoc/release-4.2.1.adoc]:
```suggestion
{productname} did not detect existing metrics-server installed by the administrator, so please remove the installed metrics-server manually before upgrade addons.
```

This is missing the commands needed to remove this. I think it also should go into the "required actions" part of the notes.

Will not removing the old metrics-server break the functionality? If so we need to specifically add this to the upgrade instructions as well.

**Comment by r0ckarong** [comment]
@jordimassaguerpla We will update this with 4.2.3 right?
@jenting Does this also apply to 4.5.x?

**Comment by jenting** [comment]
> @jordimassaguerpla We will update this with 4.2.3 right?
> 
> @jenting Does this also apply to 4.5.x?

Basically, the metrics-server released at 4.2 if I remember correctly, so before user upgrade to 4.5, the admin needs to upgrade to 4.2 first from our upgrade mechanism. So, I think written in 4.2 is enough.

**Comment by jordimassaguerpla** [comment]
> @jordimassaguerpla We will update this with 4.2.3 right?
> @jenting Does this also apply to 4.5.x?

4.2.4

**Comment by jordimassaguerpla** [comment]
> > @jordimassaguerpla We will update this with 4.2.3 right?
> > @jenting Does this also apply to 4.5.x?
> 
> 4.2.4

4.5.2

**Review by innobead — APPROVED** [review]

File: adoc/release-4.2.1.adoc (hunk 1 above)
Comment (jenting) [reply adoc/release-4.2.1.adoc]: > Will not removing the old metrics-server break the functionality? If so we need to specifically add this to the upgrade instructions as well.

I think yes, let me update this PR later~

Comment (jenting) [reply adoc/release-4.2.1.adoc]: @r0ckarong updated! PTAL :bow:

**Comment by r0ckarong** [comment]
> > @jordimassaguerpla We will update this with 4.2.3 right?
> > @jenting Does this also apply to 4.5.x?
> 
> 4.2.4

I think this should be part of the 4.5.x release notes too? If the user is on 4.2.x and wants to to an upgrade to 4.5 they need to run addon upgrade, no? If they have metrics-server installed on their existing installation they will run into that problem.

**Comment by jenting** [comment]
> > > @jordimassaguerpla We will update this with 4.2.3 right?
> 
> > > @jenting Does this also apply to 4.5.x?
> 
> > 
> 
> > 4.2.4
> 
> 
> 
> I think this should be part of the 4.5.x release notes too? If the user is on 4.2.x and wants to to an upgrade to 4.5 they need to run addon upgrade, no? If they have metrics-server installed on their existing installation they will run into that problem.

If the user's cluster is on 4.2, then it have the metrics-server deployed by skuba add-on, not deployed manually by helm. So, I think it would happens from 4.2 to 4.5 upgrade.

**Comment by r0ckarong** [comment]
> > > > @jordimassaguerpla We will update this with 4.2.3 right?
> > 
> > 
> > > > @jenting Does this also apply to 4.5.x?
> > 
> > 
> > > 
> > 
> > 
> > > 4.2.4
> > 
> > 
> > I think this should be part of the 4.5.x release notes too? If the user is on 4.2.x and wants to to an upgrade to 4.5 they need to run addon upgrade, no? If they have metrics-server installed on their existing installation they will run into that problem.
> 
> If the user's cluster is on 4.2, then it have the metrics-server deployed by skuba add-on, not deployed manually by helm. So, I think it would happens from 4.2 to 4.5 upgrade.

I don't fully understand the scenario. When would the user install metrics-server via helm? If this was an addon in 4.2.x did we stop shipping this as an addon and now you must install via helm?

**Pull request merged**